### ISSUE TITLE
Naming Conflict with cProfile Causes ImportError in zipformer/profile.py

### ISSUE THREAD

**Issue by yfyeung** [k2-fsa/icefall#1423]
While running a script that imports modules from the zipformer library, I encountered a naming conflict with Python's built-in cProfile module. It seems that the script attempts to import profile as _pyprofile from cProfile.py, but mistakenly calls zipformer/profile.py instead, which leads to an Exception: call profile.py wrongly.

Steps to Reproduce:
```shell
git clone https://github.com/k2-fsa/icefall.git
cd icefall/egs/librispeech/ASR
python zipformer/train.py
```

```shell
./train.sh: Line 8: 8 847 Aborted
（core dumped） ./zipformer/train.py --world-size 1 --num-epochs 40 --start-epoch 1 --use-fp16 1 --exp-dir zipformer/exp --full-libri 1 --max-duration 1000
```

Actual Behavior:
The script raises an Exception due to the incorrect import caused by the naming conflict.

Possible Solution:
Consider renaming zipformer/profile.py to avoid naming collisions with Python's standard library modules.

Environment:
https://github.com/k2-fsa/icefall/blob/master/docker/torch2.1.0-cuda11.8.dockerfile

**Comment by csukuangfj** [comment]
Thanks for reporting!

How about renaming it to `profiling.py`?


---

We need to add CI tests to cover train.py

**Comment by csukuangfj** [comment]
Fixed by #1424
